echo/app/src/main/java/com/google/sample/echo/MainActivity.java

```diff
@@ -0,0 +1,132 @@
+package com.google.sample.echo;
+
+import android.app.Activity;
+import android.content.Context;
+import android.media.AudioManager;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.TextView;
+
+public class MainActivity extends Activity {
+    public static final String AUDIO_SAMPLE = "AUDIO_SAMPLE:";
+    TextView status_view;
+    String  nativeSampleRate;
+    String  nativeSampleBufSize;
+    String  nativeSampleFormat;
+    Boolean isPlaying;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        status_view = (TextView)findViewById(R.id.statusView);
+        queryNativeAudioParameters();
+
+        // initialize native audio system
+        updateNativeAudioUI();
+        createSLEngine(Integer.parseInt(nativeSampleRate), Integer.parseInt(nativeSampleBufSize));
+        isPlaying = false;
+    }
+    @Override
+    protected void onDestroy() {
+        if( isPlaying ) {
+            stopPlay();
+        }
+        //shutdown();
+        deleteSLEngine();
+        isPlaying = false;
+        super.onDestroy();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    public void StartEcho(View view) {
+        status_view.setText("StartCapture Button Clicked\n");
+        if(isPlaying) {
+            return;
+        }
+        if(!createSLBufferQueueAudioPlayer()) {
```

Comment:
It might be better to keep the JNI binding minimal: just a create() function, and have separate method in the EchoAudioEngine type for each element creation.